examples/servers/nginx/devbox.d/nginx/nginx.template

```diff
@@ -0,0 +1,20 @@
+events {}
+http{
+server {
+         listen       $NGINX_WEB_PORT;
+         listen       [::]:$NGINX_WEB_PORT;
+         server_name  localhost;
+         root         ../../../devbox.d/web;
```

Comment:
Should `server_name` and `root` be configurable through environment variables too?

Reply:
Adding variables for both of these

Reply:
Added the following to the plugin: 

`WEB_PORT`
`WEB_SERVER_NAME`
`WEB_ROOT`

These are all set by the plugin, and can be overridden using the `.env` support or via the `env` section in the devbox.json. `devbox services up|start|restart` or will automatically regenerate the `nginx.conf` using: 

```
envsubst < $NGINX_CONFDIR/nginx.template > $NGINX_CONFDIR/nginx.conf
```

If for whatever reason you want to lock the `nginx.conf`, you can delete the `nginx.template` file.